iothub_service_client/src/iothub_service_client_auth.c

```diff
@@ -111,13 +166,20 @@ IOTHUB_SERVICE_CLIENT_AUTH_HANDLE IoTHubServiceClientAuth_CreateFromConnectionSt
                         free_service_client_auth(result);
                         result = NULL;
                     }
-                    else if ((sharedAccessKey = Map_GetValueFromKey(connection_string_values_map, IOTHUBSHAREDACESSKEY)) == NULL)
+                    else if (!useSharedAccessSignature && (sharedAccess = Map_GetValueFromKey(connection_string_values_map, IOTHUBSHAREDACESSKEY)) == NULL)
                     {
                         /*Codes_SRS_IOTHUBSERVICECLIENT_12_013: [** If the populating SharedAccessKey fails, IoTHubServiceClientAuth_CreateFromConnectionString shall do clean up and return NULL. **] */
                         LogError("Couldn't find %s in connection string", IOTHUBSHAREDACESSKEY);
                         free_service_client_auth(result);
                         result = NULL;
                     }
+                    else if (useSharedAccessSignature && (sharedAccess = Map_GetValueFromKey(connection_string_values_map, IOTHUBSHAREDACESSSIGNATURE)) == NULL)
+                    {
+                        /*Codes_SRS_IOTHUBSERVICECLIENT_12_013: [** If the populating SharedAccessKey fails, IoTHubServiceClientAuth_CreateFromConnectionString shall do clean up and return NULL. **] */
```

Comment:
>IoTHubServiceClientAuth_CreateFromConnectionString [](start = 109, length = 50)

update method name to IoTHubServiceClientAuth_CreateFromSharedAccessSignature

Reply:
ok